vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -8,18 +8,21 @@ class SecureRoomsApi::ScansController < ApplicationController
   before_action :load_user_and_reader
 
   def scan
-    accounts = @user.accounts_for_product(@card_reader.secure_room)
-    requested_id = params[:account_identifier].to_i
-    selected_account = accounts.find { |account| account.id == requested_id }
+    requested_account_id = params[:account_identifier].to_i
 
-    access_verdict = SecureRooms::CheckAccess.new.authorize(
+    verdict = SecureRooms::CheckAccess.new.authorize(
       @user,
       @card_reader,
-      accounts: accounts,
-      selected_account: selected_account,
+      requested_account_id: requested_account_id,
     )
 
-    render SecureRooms::ScanResponsePresenter.new(@user, @card_reader, access_verdict, accounts).response
+    SecureRooms::AccessManager.new(
```

Comment:
I think they do each deserve to be their own object, but the usage of these three objects still feels a little funny to me. Maybe because the `AccessManager` is now acting more as a callback vs actually managing anything. Maybe it needs a better name? Maybe we hold off on seeing what else it does before we focus on the better name.

we could do an api like`SecureRooms::AccessManager.handle(verdict)`? I sometimes like using class methods like that to improve the API.